### PR TITLE
Adding tests for the get_authors function

### PR DESCRIPTION
**Pull request by ialarmedalien**
# Description of PR purpose/changes

- Fixed the broken `get_authors` function and added tests.

# Jira Ticket / Issue #

Related Jira ticket: https://kbase-jira.atlassian.net/browse/DATAUP-X
- [ ] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
- [x] Tests pass locally and in GitHub Actions

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x ] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] (Python) I have run Black and Flake8 on changed Python code
- [ ] If appropriate, I have recompiled the app and added the updated `StaticNarrativeImpl.py`, `StaticNarrativeServer.py`, and `compile_report.json` to this PR.

# Updating Version and Release Notes (if applicable)

- [ ] [Version has been bumped](https://semver.org/) for each release
- [ ] [Release notes](/RELEASE_NOTES.md) have been updated for each release (and during the merge of feature branches)

### CODE REVIEW

File: lib/StaticNarrative/exporter/processor_util.py [hunk 1]
```diff
@@ -251,20 +251,35 @@ def get_data_icon(obj_type: str) -> dict[str, str]:
...
+        author
+        for author in other_authors
+        if other_authors[author] in ["w", "a"] and author != "*" and author != ws_author
+    }
 
-    auth = KBaseAuth(config.auth_url)
+    author_id_list = [ws_author, *sorted(author_id_set)]
     disp_names = {}
     try:
-        disp_names = auth.get_display_names(config.token, author_id_list)
```
Comment (ialarmedalien) [inline lib/StaticNarrative/exporter/processor_util.py:267]: KBaseAuth client doesn't have this method so it was failing every time.

File: lib/StaticNarrative/exporter/processor_util.py [hunk 2]
```diff
@@ -251,20 +251,35 @@ def get_data_icon(obj_type: str) -> dict[str, str]:
...
 
-def get_authors(ws_client: Workspace, config: Config, wsid: str) -> list[dict[str, str]]:
+def get_display_names(auth_url: str, token: str, user_ids: list[str]) -> dict[str, Any]:
+    """Given a list of user IDs, get the corresponding display names."""
+    headers = {"Authorization": token}
+    r = requests.get(
+        f"{auth_url}/api/V2/users/?list=" + ",".join(user_ids),
+        headers=headers,
+        timeout=30 * 60,  # same as baseclient default
+    )
```
Comment (ialarmedalien) [inline lib/StaticNarrative/exporter/processor_util.py:261]: Just use requests to access the auth API directly (this is how the narrative does it, too).

File: test/mocks.py [hunk 3]
```diff
@@ -33,7 +33,9 @@ def _mock_adapter(
     :param ws_info: list - the Workspace info that should be returned on a call to
         Workspace.get_workspace_info
     :param ws_perms: dict - a mapping from a workspace id to a dict with user permissions on that ws
-    :param user_map: dict - a mapping from user id to full name, used in calls to Auth
+        Workspace.get_permissions
+    :param user_map: dict - a mapping from user id to full name, used in calls to Auth.
+        If the user map is an empty dictionary, the endpoint will return an error.
```
Comment (ialarmedalien) [inline test/mocks.py:38]: added this in for testing purposes

File: test/test_processor_util.py [hunk 4]
```diff
@@ -218,3 +222,77 @@ def test_get_created_objects_from_report(ws_client: Workspace, params: dict[str,
...
+
+
+USERS = ["owner", "user_1", "user_2", "user_3"]
+NAMES = ["The Wizard of Oz", "<insert name here>", "Me; Myself; & I", "Occam's Razor"]
+HTML_NAMES = [
+    "The Wizard of Oz",
+    "&lt;insert name here&gt;",
+    "Me; Myself; &amp; I",
+    "Occam's Razor",
+]
```
Comment (ialarmedalien) [inline test/test_processor_util.py:234]: names are html-escaped - hence the use of `<`, `>`, and `&` in these names.

File: lib/StaticNarrative/exporter/processor_util.py (hunk 1 above)
Comment (briehl) [reply lib/StaticNarrative/exporter/processor_util.py:267]: D'oh!

**Review by briehl — APPROVED** [review]

**Pull request merged**